diff --git a/mocha.js b/mocha.js
index 21ed960..75f5a9d 100644
--- a/mocha.js
+++ b/mocha.js
@@ -194,9 +194,13 @@ define(function(require, exports, module) {
                         return (debugging = false); // Either the debugger is not found or paused
                 });
             }
-            
+
             var path = join(c9.workspaceDir, fileNode.path);
+
             args.push(path);
+
+            var projectDir = /^\/[^\/]+(?=\/)/.exec(fileNode.path);
+            projectDir = join(c9.workspaceDir, projectDir ? projectDir[0] : '/');
             
             var coveragePath = "~/.c9/coverage/run" + (++uniqueId);
             if (withCodeCoverage) {
@@ -214,10 +218,10 @@ define(function(require, exports, module) {
                 args.unshift("-l", "-c", '"$0" "$@"', exec); // force a login shell
                 exec = "bash";
             }
-            
+
             proc.pty(exec, {
                 args: args,
-                cwd: dirname(path),
+                cwd: projectDir,
                 fakePty: isWin
             }, function(err, pty){
                 if (err) return callback(err);
@@ -607,4 +611,4 @@ define(function(require, exports, module) {
             "test.mocha": plugin
         });
     }
-});
\ No newline at end of file
+});
